Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1185,16 +1185,20 @@ static String mutationObserverNotificationString()
 void UnifiedPDFPlugin::updateLayout(AdjustScaleAfterLayout shouldAdjustScale)
 {
     auto layoutSize = availableContentsRect().size();
-    m_documentLayout.updateLayout(layoutSize, m_shouldUpdateAutoSizeScale);
+
+    auto autoSizeMode = m_didLayoutWithValidDocument ? m_shouldUpdateAutoSizeScale : ShouldUpdateAutoSizeScale::Yes;
+    m_documentLayout.updateLayout(layoutSize, autoSizeMode);
     updateScrollbars();
 
     // Do a second layout pass if the first one changed scrollbars.
     auto newLayoutSize = availableContentsRect().size();
     if (layoutSize != newLayoutSize) {
-        m_documentLayout.updateLayout(newLayoutSize, m_shouldUpdateAutoSizeScale);
+        m_documentLayout.updateLayout(newLayoutSize, autoSizeMode);
         updateScrollbars();
     }
 
+    m_didLayoutWithValidDocument = m_documentLayout.havePDFDocument();
```

Comment:
Should we also change the auto size state change underneath this to only set it to `No` if we did a layout with a valid document?

Reply:
(actually probably not due to the same reason above)